docs/concepts/simulation.md

```diff
@@ -0,0 +1,72 @@
+The goal of this is to detect and debug failures that could halt a live chain, by providing logs and statistics about the operations run by the simulator as well as exporting the latest application state when a failure was found.
+
+Its main difference with integration testing is that the simulator app allows you to pass parameters to customize the chain that's being simulated.
+This comes in handy when trying to reproduce bugs that were generated in the provided operations (randomized or not).
+
+## Simulation commands
+
+The simulation app has different commands, each of it tests a different failure type:
+
+- `FullAppSimulation`: General simulation mode. Runs the chain and the specified operations for a given number of blocks. Tests that there're no `panics` on the simulation.
+- `AppStateDeterminism`: Checks that all the nodes return the same values, in the same order.
+- `AppImportExport`: The simulator exportes the initial app state and then it creates a new app with the exported `genesis.json` as an input, checking for inconsistencies between the stores.
+- `AppSimulationAfterImport`: Queues two simulations together. The first one provides the app state (_i.e_ genesis) to the second. Useful to test software upgrades or hard-forks from a live chain.
+- `BenchmarkInvariants`: An invariant checks for differences between the values that are on the store and the passive tracker. Eg: total coins held by accounts vs total supply tracker.
+
+Each simulation must receive a set of inputs (_i.e_ flags) such as the number of blocks that the simulation is run, seed, block size, etc.
+Check the full list of flags [here](https://github.com/cosmos/cosmos-sdk/blob/adf6ddd4a807c8363e33083a3281f6a5e112ab89/simapp/sim_test.go#L34-L50).
+
+## Simulator Modes
+
+In addition to the various inputs and commands. The simulator runs in three modes:
+
+1. Completely random where the initial state, module parameters and simulation parameters are **pseudo-randomly generated**.
+2. From a `genesis.json` file where the initial state and the module parameters are defined.
+This mode is helpful for running simulations on a known state such as a live network export where a new (mostly likely breaking) version of the application needs to be tested.
+3. From a `params.json` file where the initial state is pseudo-randomly generated but the module and simulation parameters can be provided manually.
+This allows for a more controlled and deterministic simulation setup while allowing the state space to still be pseudo-randomly simulated. The list of available parameters is listed [here](https://github.com/cosmos/cosmos-sdk/blob/adf6ddd4a807c8363e33083a3281f6a5e112ab89/x/simulation/params.go#L170-L178).
+
+::: tip
+These modes are not excluding between each other. So you can for example run a randomly generated genesis state (`1`) with manually generated simulation params (`3`).
+:::
+
+## Usage
```

Comment:
The Usage section is a bit light. I think it'd be cool to have a more in-depth look at how to run a simulation for your own SDK application.

As a developer, do I have to prepare my application a certain way before I can use the simulator? (e.g. include `simApp`, make `app.go` a certain way, etc.). Once it's done, what do I need to to?

Reply:
I agree with all your comments here but I think it's better to point to the tutorial once it's updated to resolve those questions. What do you think ?

This section rn is mostly for how to run the simulator

Reply:
Agree with @fedekunze, I think we solve for that by adding it to the tutorial. I think @hschoenburg said he was working on some advanced tutorial content.